unity-renderer/Assets/DCLPlugins/ECS7/InternalECSComponents/InternalECSComponents.cs

```diff
@@ -66,4 +83,28 @@ public void Dispose()
         renderersComponent.Dispose();
         inputEventResultsComponent.Dispose();
     }
-}
+
+    internal static Action WriteSystem(IList<InternalComponentWriteData> scheduledWrite)
+    {
+        return () =>
+        {
+            for (int i = 0; i < scheduledWrite.Count; i++)
+            {
+                var writeData = scheduledWrite[i];
+                if (writeData.scene == null)
+                    continue;
+
+                InternalComponent data = writeData.data;
+                if (data != null)
+                {
+                    data._dirty = false;
+                }
+                else
+                {
+                    writeData.scene.crdtExecutor.ExecuteWithoutStoringState(writeData.entityId, writeData.componentId, null);
+                }
+            }
+            scheduledWrite.Clear();
+        };
+    }
+}
```

Comment:
This is super strange.

Why cannot `scheduleWrite` be a class variable `WriteSystem` a parameterless void method in the interface?

It will also create garbage due to the closure using a local scoped variable.

Reply:
dependency injection.
not having the method tied to a class state makes room for easier, cleaner and better tests
and at the same time it can be tied to the class state in it's implementation. 
anyway, I applied the suggested change here https://github.com/decentraland/unity-renderer/pull/3143/commits/685c90c25fec6e127d53bae91d7c3446c72bfd15